datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -201,15 +216,26 @@ public static V1Deployment getApiDeployment(String namespace) throws ApiExceptio
    */
   public static V1Deployment changeReplicaSetSize(V1Deployment deployment, int numberOfReplicas)
       throws ApiException {
-    V1DeploymentSpec existingSpec = deployment.getSpec();
-    deployment.setSpec(existingSpec.replicas(numberOfReplicas));
-    return kubernetesClientAppsObject.replaceNamespacedDeployment(
-        deployment.getMetadata().getName(),
-        deployment.getMetadata().getNamespace(),
-        deployment,
-        null,
-        null,
-        null);
+    V1Deployment deploy;
+    try {
+      V1DeploymentSpec existingSpec = deployment.getSpec();
+      deployment.setSpec(existingSpec.replicas(numberOfReplicas));
+      deploy =
+          kubernetesClientAppsObject.replaceNamespacedDeployment(
+              deployment.getMetadata().getName(),
+              deployment.getMetadata().getNamespace(),
+              deployment,
+              null,
+              null,
+              null);
+    } catch (ApiException ex) {
+      System.out.println(
+          "Scale the pod failed for Deployment. Exception: "
+              + ex.getMessage()
+              + ex.getStackTrace());
```

Comment:
This seems out of place. Was an exception here not bubbling up to the top level when you called the TestRunner main method?

Reply:
Ok - I just tested and with your latest changes, it is bubbling up now without this catch. It should be removed!